pyscript.core/test/worker/worker.rb

```diff
@@ -1,10 +1,6 @@
-require "js"
-
-xworker = JS::eval("return xworker")
-
 def on_message(event)
   puts event[:data]
-  xworker.postMessage('Ruby: Hello MicroPython 👋')
+  $xworker.call('postMessage', 'Ruby: Hello MicroPython 👋')
 end
 
-xworker.onmessage = on_message
+$xworker[:onmessage] = -> (event) { on_message event }
```

Comment:
Does this means that we now need to do `$worker` to say 'run this in worker?

Reply:
in *Ruby* a global is accessed via `$` prefix ... this is *Ruby* code, and that's needed because of the previous lack of `registerJSModule` I've mentioned, so until we can import `xworker` in *Ruby*, this is needed.

Once again, *Ruby* code here, no other interpreter needs the *Ruby* convention, here we run *Ruby* and I believe *Ruby* developers know about `$globals` being defined as such.

In *Python* we `import xworker from xworker` (module name soon to change though) and in *Lua* (Wasmoon) you just use `xworker` right away.

Note that this *Ruby* code also can't assign just the method, it needs an ugly lambda indirection, this is all part of *Ruby* being experimental, the rest of the interpreters are way more mature and working in a way more natural way.

Reply:
Sounds good was just curious 😄